libcudacxx/include/cuda/std/utility

```diff
@@ -57,6 +57,7 @@
 #include <cuda/std/__utility/swap.h>
 #include <cuda/std/__utility/to_underlying.h>
 #include <cuda/std/__utility/unreachable.h>
+#include <cuda/std/__variant/monostate.h>
```

Comment:
I would like to properly move the file from `__utility` to `__variant` so that it is clear where it lives

Reply:
I've also extended the testing to all compilers and C++11 as I see no reason why it shouldn't work.